api/pkg/transformer/jsonpath/jsonpath.go

```diff
@@ -1,18 +1,28 @@
 package jsonpath
 
 import (
+	"context"
 	"fmt"
 	"reflect"
 	"regexp"
 
-	"github.com/gojekfarm/jsonpath"
-
+	"github.com/caraml-dev/protopath"
 	"github.com/gojek/merlin/pkg/transformer/spec"
 	"github.com/gojek/merlin/pkg/transformer/types"
 	"github.com/gojek/merlin/pkg/transformer/types/converter"
+	"github.com/gojekfarm/jsonpath"
+	"google.golang.org/protobuf/proto"
 )
 
+// SourceType indicates type of source for jsonpath value extraction
+type SourceType int
+
 const (
+	// Map source type is in map type
+	Map SourceType = iota
```

Comment:
This const declaration is a bit confusing, can `Map` and `Proto` be separated from other const?